### PR TITLE
Port some code that we use from github.com/Azure/aks-engine/pkg/helpers.

### PR DESCRIPTION
**Pull request by yizhang4321**
This removes the explicit dependency on github.com/Azure/aks-engine/pkg/helpers for some of our code. Our end goal is to remove all aks-engine dependencies.

### CODE REVIEW

File: pkg/aks-engine/helpers/azureconst.go [hunk 1]
```diff
@@ -0,0 +1,1698 @@
...
+		"usgovvirginia",
+		"usgoviowa",
+		"usgovarizona",
+		"usgovtexas",
+		"francecentral",
+	}
+}
+
+// GetDCOSMasterAllowedSizes returns the master allowed sizes
+func GetDCOSMasterAllowedSizes() string {
```
Comment (xuto2) [inline pkg/aks-engine/helpers/azureconst.go:76]: we dont need this

File: pkg/aks-engine/helpers/azureconst.go [hunk 2]
```diff
@@ -0,0 +1,1698 @@
...
+        "Standard_NV6s_v2",
+        "Standard_PB12s",
+        "Standard_PB24s",
+        "Standard_PB6s"
+    ],
+`
+}
+
+// GetKubernetesAllowedVMSKUs returns the allowed sizes for Kubernetes agent
+func GetKubernetesAllowedVMSKUs() string {
```
Comment (xuto2) [inline pkg/aks-engine/helpers/azureconst.go:387]: do we need this?

File: pkg/aks-engine/helpers/helpers.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+// GetTargetEnv determines and returns whether the region is a sovereign cloud which
+// have their own data compliance regulations (China/Germany/USGov) or standard
+// Azure public cloud
+// CustomCloudName is name of environment if customCloudProfile is provided, it will be empty string if customCloudProfile is empty.
+// Because customCloudProfile is empty for deployment for AzurePublicCloud, AzureChinaCloud,AzureGermanCloud,AzureUSGovernmentCloud,
+// The customCloudName value will be empty string for those clouds
+func GetTargetEnv(location, customCloudName string) string {
+	switch {
+	case customCloudName != "" && strings.EqualFold(customCloudName, "AzureStackCloud"):
```
Comment (xuto2) [inline pkg/aks-engine/helpers/helpers.go:101]: we don't need this

File: pkg/aks-engine/helpers/azureconst.go [hunk 4]
```diff
@@ -0,0 +1,1698 @@
...
+        "Standard_NV6s_v2",
+        "Standard_PB12s",
+        "Standard_PB24s",
+        "Standard_PB6s"
+    ],
+`
+}
+
+// GetSizeMap returns the size / storage map
+func GetSizeMap() string {
```
Comment (xuto2) [inline pkg/aks-engine/helpers/azureconst.go:719]: We don't need this

File: pkg/aks-engine/helpers/azureconst.go [hunk 5]
```diff
@@ -0,0 +1,1698 @@
...
+// Licensed under the MIT license.
+
+package helpers
+
+// AUTOGENERATED FILE
+
+// GetAzureLocations provides all azure regions in prod.
+// Related powershell to refresh this list:
+//   Get-AzureRmLocation | Select-Object -Property Location
+func GetAzureLocations() []string {
```
Comment (xuto2) [inline pkg/aks-engine/helpers/azureconst.go:11]: where do we use this? 

I feel we don't need this azureconst.go at all.

File: pkg/aks-engine/helpers/helpers.go (hunk 3 above)
Comment (yizhang4321) [reply pkg/aks-engine/helpers/helpers.go:101]: Used in pkg/agent/datamodel/types.go:
targetEnv := helpers.GetTargetEnv(cs.Location, cs.Properties.GetCustomCloudName())

But maybe we can remove that code in pkg/agent/datamodel/types.go as well?

File: pkg/aks-engine/helpers/azureconst.go (hunk 1 above)
Comment (yizhang4321) [reply pkg/aks-engine/helpers/azureconst.go:76]: Removed.

File: pkg/aks-engine/helpers/azureconst.go (hunk 2 above)
Comment (yizhang4321) [reply pkg/aks-engine/helpers/azureconst.go:387]: Removed.

File: pkg/aks-engine/helpers/azureconst.go (hunk 4 above)
Comment (yizhang4321) [reply pkg/aks-engine/helpers/azureconst.go:719]: Removed.

File: pkg/aks-engine/helpers/azureconst.go (hunk 5 above)
Comment (yizhang4321) [reply pkg/aks-engine/helpers/azureconst.go:11]: The only thing left in azureconst.go is GetAzureLocations(), which is currently used pkg/aks-engine/engine/output.go and pkg/agent/datamodel/types.go. Maybe we don't need those code that uses it, too? I'll keep it for now and do another pass in the future. There are definitely more code than this that we don't need.

**Pull request merged**